test/AspNet.Security.OAuth.Providers.Tests/LinkedIn/LinkedInTests.cs

```diff
@@ -50,5 +55,33 @@ public async Task Can_Sign_In_Using_LinkedIn(string claimType, string claimValue
                 AssertClaim(claims, claimType, claimValue);
             }
         }
+
+        [Theory]
+        [InlineData(ClaimTypes.NameIdentifier, "1R2RtA")]
+        [InlineData(ClaimTypes.Name, "Frodon Sacquet")]
+        [InlineData(ClaimTypes.GivenName, "Frodon")]
+        [InlineData(ClaimTypes.Surname, "Sacquet")]
+        public async Task Can_Sign_In_Using_LinkedIn_Localized(string claimType, string claimValue)
+        {
+            // Arrange
+            additionnalConfiguration = options => options.MultiLocaleStringResolver = (values, preferredLocale) =>
```

Comment:
Could also add a second test method case here where a `try-finally` block is used to change the current thread culture and test that the default using the thread's culture also correctly returns the French version?

Reply:
In fact, it is not the test thread culture that should be changed, but the test server one.
So I added a virtual method on `OAuthTests<TOptions>` to register additional middlewares, and implemented it in `LinkedInTests` to add `UseRequestLocalization` on the application builder.